File: q0441/q441.cpp
class Solution
{
public:
    int arrangeCoins(int n)
    {
        int l = 0;
        int r = n;
        while (l <= r)
        {
            long m = (l + r) / 2;
            /* set l to be m+1 even if we find m, this is because we want to find the largest m possible */
            if (m * (m + 1) / 2 <= n)
            {
                l = m + 1;
            }
            else
            {
                r = m - 1;
            }
        }
        return l - 1;
    }
};

File: q0441/q441.py
class Solution:
    def arrangeCoins(self, n: int) -> int:
        l = 0
        r = n
        while l <= r:
            m = (l + r) // 2
            if m * (m + 1) // 2 <= n:
                l = m + 1
            else:
                r = m - 1
        return l - 1
